base/intfuncs.jl

```diff
@@ -87,13 +92,16 @@ end
 gcdx(a::Integer, b::Integer) = gcdx(promote(a,b)...)
 
 # multiplicative inverse of n mod m, error if none
-function invmod(n, m)
+function invmod{T<:Integer}(n::T, m::T)
     g, x, y = gcdx(n, m)
-    if g != 1 || m == 0
-        error("no inverse exists")
-    end
-    x < 0 ? abs(m) + x : x
+    (g != 1 || m == 0) && throw(DomainError())
+    # Note that m might be negative here.
+    # For unsigned T, x might be close to typemax; add m to force a wrap-around.
```

Comment:
Good question. I did not prove this. The previous implementation says `abs(m) + x`, indicating that adding `m` suffices. I also tested for all 8-bit integers where there was no overflow in intermediate results, and adding `m` was sufficient there. The conditions guaranteed by the `gcdx` algorithm indicate that the result `r` will be "near zero" (see the `gcdx` doc string), so I think this is fine.